Libraries/NativeAnimation/Nodes/RCTValueAnimatedNode.h

```diff
@@ -24,6 +24,7 @@
 - (void)extractOffset;
 
 @property (nonatomic, assign) CGFloat value;
```

Comment:
I think we should change the type of value to be `id` on iOS or `Object` on Android. Then we can use instanceof checks in when we need to use the actual value.

Reply:
Is Object preferable to Dynamic here? Both require boxing of primitives, but Object is probably cheaper anyway. Just not completely sure how this would interact with other native libraries for react-native, or how many of them depend on this being a primitive data type. Also the cost of boxing can be quite significant and can even cause heap allocations. A float would take 4 bytes normally, but a Float can be 16(-ish) bytes, plus the cost of the pointer (device/cpu dependent). I'm afraid this could cause a noticeable performance regression in some cases.

Reply:
I just realised that the JavaOnlyMap will cause autoboxing of the double into a Double already anyway (and the android implementation already uses doubles rather than floats). So using Object should be fine.

Reply:
@janicduplessis How about keeping value as a double, and adding another property called objValue? This can be used for general purposes such as String interpolation. Changing all the uses of value to have dynamic type will probably come with a noticeable performance impact for the current Animated use cases. Having it isolated to a single if statement, like this pr has it, but changing it to Object and using instanceof to handle the String case, seems like the best tradeoff for me atm. What do you think?

Reply:
Renamed stringValue to animatedObject, and changed String to id in obj-c and Object in java https://github.com/facebook/react-native/pull/18187/commits/0521b3e085864d632cb257310c4b34195fab92f3